modules/structs/repo_branch.go

```diff
@@ -41,6 +41,7 @@ type BranchProtection struct {
 	BlockOnRejectedReviews      bool     `json:"block_on_rejected_reviews"`
 	DismissStaleApprovals       bool     `json:"dismiss_stale_approvals"`
 	RequireSignedCommits        bool     `json:"require_signed_commits"`
+	ProtectedFilePatterns       string   `json:"protected_file_patterns"`
```

Comment:
Just a comment: Isn't this a little excessive? json.Marshal and Unmarshal would give most of these the same names anyway

Reply:
No it would not. Default golang json serialization would create field `protectedFilePatterns`